bin/compare-factory-config.py

```diff
@@ -122,6 +132,34 @@ def remove_readonly(func, path, _):
     func(path)
 
 
+def find_non_matches(gfactory_DB, topology_DB, mode):
+    ret = []
+    # ResourceNames that does not match any resource records in TopologyDB
+    # they may have match in other tags, or not in TopologyDB
+    if (mode == 'resources'):
+        # GLIDEIN_ResourceNames that does not match resources records in TopologyDB
+        nonmatch_resource_names = set(gfactory_DB.keys()).difference(
+            topology_DB['resources'])
+        # Factory ResourceNames that match TopologyDB's entries other than a resource
+        match_non_resource_names = nonmatch_resource_names.intersection(
+            topology_DB['resourceGroups'].union(
+                topology_DB['sites'], topology_DB['facilities']))
+
+        for name in gfactory_DB.keys():
+            if name in match_non_resource_names:
+                for entry in gfactory_DB[name]:
+                    ret.append((entry, name))
```

Comment:
```suggestion
        for name in match_non_resource_names:
            for entry in gfactory_DB[name]:
                ret.append((entry, name))
```

As written, `match_non_resource_names` is a subset of `gfactory_db.keys()` so the `if` is superfluous and you'll want to loop over the smaller set

Reply:
good catch!